packages/ctl/ctl.js

```diff
@@ -903,6 +903,26 @@ userCommands.add(new libCommand.Command({
 	},
 }));
 
+userCommands.add(new libCommand.Command({
+	definition: ["admin-bulk-import <file>", "Bulk input from file users to be admin", (yargs) => {
+		yargs.positional("file", { describe: "File to import from", type: "string" });
+		yargs.options({
+			"create": { describe: "Create user if it does not exist", nargs: 0, type: "boolean", default: false },
+		});
+	}],
+	handler: async function(args, control) {
+		let data = await fs.readFile(args.file, "utf8");
+		let names = data.split("\n");
+		await names.forEach(async (name) => {
+			if (name !== "") {
+				await libLink.messages.setUserAdmin.send(control, {
+					name: name, create: args.create, admin: true,
+				});
+			}
+		});
```

Comment:
This does not do what you think it does. If you're using a function you're unsure of how it works [consult a reference for it](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach). `forEach` returns `undefined` awaiting it is a logical error here. The callback is not awaited which means all of the requests will be started concurrently and that if any one of them throws Node.js will abort with an unhandled promise rejection. A regular `for (... of ...)` loop would be more appropriate here.